core/src/instance.rs

```diff
@@ -149,45 +138,46 @@ impl Default for Instance {
 }
 
 /// Send Action to Instance's Event Queue and block until is has been processed.
+///
+/// panics if called before start_action_loop
 pub fn dispatch_action_and_wait(
     action_channel: &Sender<ActionWrapper>,
     observer_channel: &Sender<Observer>,
-    action_wrapper: &ActionWrapper,
+    action_wrapper: ActionWrapper,
 ) {
     // Create blocking channel
-    let (sender, receiver) = channel::<bool>();
+    let (sender, receiver) = channel::<()>();
```

Comment:
We don't actually care what value gets sent so it's the perfect use for the unit type.